lib/Config.php

```diff
@@ -4,13 +4,30 @@
 
 final class Config
 {
+    /** @var array */
     private $nameservers;
+    /** @var array */
     private $knownHosts;
+    /** @var int */
     private $timeout;
+    /** @var int */
     private $attempts;
-
-    public function __construct(array $nameservers, array $knownHosts = [], int $timeout = 3000, int $attempts = 2)
-    {
+    /** @var array */
+    private $searchList;
+    /** @var int */
+    private $ndots;
+    /** @var bool */
+    private $rotate;
+
+    public function __construct(
+        array $nameservers,
+        array $knownHosts = [],
+        int $timeout = 3000,
+        int $attempts = 2,
+        array $searchList = [],
+        int $ndots = 1,
+        bool $rotate = false
```

Comment:
I'm not sure whether we should continue adding arguments here or instead start writing mutators like `withSearchList()` returning a new modified instance.